libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3763,13 +3763,27 @@ bool TypeChecker::visit(Identifier const& _identifier)
 				"\"suicide\" has been deprecated in favour of \"selfdestruct\"."
 			);
 		else if (_identifier.name() == "selfdestruct" && fType->kind() == FunctionType::Kind::Selfdestruct)
-			m_errorReporter.warning(
-				5159_error,
-				_identifier.location(),
-				"\"selfdestruct\" has been deprecated. "
-				"The underlying opcode will eventually undergo breaking changes, "
-				"and its use is not recommended."
-			);
+		{
+			if (m_evmVersion <= langutil::EVMVersion::shanghai())
```

Comment:
It's `cancun` that's relevant here so I'd rather base the comparison on it:

```suggestion
			if (m_evmVersion < langutil::EVMVersion::cancun())
```

Reply:
I unified the warnings as you suggested here: https://github.com/ethereum/solidity/pull/14785#discussion_r1457949899 so this condition is not needed anymore.